central/graphql/resolvers/image_vulnerabilities.go

```diff
@@ -101,3 +230,313 @@ func withImageCveTypeFiltering(q string) string {
 	return search.AddRawQueriesAsConjunction(q,
 		search.NewQueryBuilder().AddExactMatches(search.CVEType, storage.CVE_IMAGE_CVE.String()).Query())
 }
+
+/*
+Sub Resolver Functions
+*/
+
+func (resolver *imageCVEResolver) EnvImpact(ctx context.Context) (float64, error) {
+	allCount, err := resolver.root.DeploymentCount(ctx, RawQuery{})
+	if err != nil || allCount == 0 {
+		return 0, err
+	}
+	scopedCount, err := resolver.root.DeploymentCount(resolver.withImageVulnerabilityScope(ctx), RawQuery{})
+	if err != nil {
+		return 0, err
+	}
+	return float64(scopedCount) / float64(allCount), nil
+}
+
+func (resolver *imageCVEResolver) FixedByVersion(ctx context.Context) (string, error) {
+	scope, hasScope := scoped.GetScope(ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_IMAGE_COMPONENTS {
+		return "", nil
+	}
+
+	edgeID := postgres.IDFromPks([]string{scope.ID, resolver.data.GetId()})
+	edge, found, err := resolver.root.ComponentCVEEdgeDataStore.Get(ctx, edgeID)
+	if err != nil || !found {
+		return "", err
+	}
+	return edge.GetFixedBy(), nil
+}
+
+func (resolver *imageCVEResolver) IsFixable(ctx context.Context, args RawQuery) (bool, error) {
+	query, err := args.AsV1QueryOrEmpty(search.ExcludeFieldLabel(search.CVEID))
+	if err != nil {
+		return false, err
+	}
+
+	conjuncts := []*v1.Query{query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery()}
```

Comment:
Make sure `query` does not have `Fixable` at this point. Ideally, we should throw an error to the user. Someone querying this field and passing a fixable query is misuse. For now, log an error because I feel UI could be passing incorrect queries and we do not want to fail.